lnwallet/channel.go

```diff
@@ -3446,27 +3445,27 @@ func (lc *LightningChannel) ProcessChanSyncMsg(
 //      it.
 //   3. We didn't get the last RevokeAndAck message they sent, so they'll
 //      re-send it.
-func (lc *LightningChannel) ChanSyncMsg() (*lnwire.ChannelReestablish, error) {
+func ChanSyncMsg(c *channeldb.OpenChannel) (*lnwire.ChannelReestablish, error) {
```

Comment:
should this have a mutex around it? esp now that this is called in more places than just the switch

also seems like something that should be implemented in channeldb as a method of `OpenChannel`, since it is otherwise unrelated to anything in `lnwallet`. would be nice to keep the `.ChanSyncMsg()` notation, and not use a package level function

Reply:
Good idea, added mutex.

The reason I didn't make this a method on `OpenChannel` is that it calls `ComputeCommitmentPoint`, which is a method from `lnwallet`, leading to an import cycle. We could however move this method to a new package (`lnutil`? 😛) if we want this, but I felt it was not worth it.

Also moving it to `channeldb` felt wrong 🤔